### PR TITLE
Added support for Istio 1.13

### PR DESCRIPTION
**Pull request by Kuvesz**
| Q               | A      |
| --------------- | ------ |
| Bug fix?        | no |
| New feature?    | no |
| API breaks?     | no |
| Deprecations?   | no |
| License         | Apache 2.0 |

### What's in this PR?
Updated Koperator to support Istio 1.13

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/developer-guide/blob/master/docs/coding-style/error-handling-guide.md)
- [x] Logging code meets the guideline

### CODE REVIEW

File: charts/kafka-operator/templates/crds.yaml [hunk 1]
```diff
@@ -978,7 +978,9 @@ spec:
                         type: string
                       description: 'Custom labels for the broker pods, example use
                         case: for Prometheus monitoring to capture the group for each
-                        broker as a label, e.g.: kafka_broker_group: "default_group"'
+                        broker as a label, e.g.: kafka_broker_group: "default_group"
+                        these labels won''t override the reserved labels that the
```
Comment (amberimam) [inline charts/kafka-operator/templates/crds.yaml:982]:
```suggestion
                        these labels won't override the reserved labels that the
```

File: charts/kafka-operator/templates/crds.yaml [hunk 2]
```diff
@@ -6874,7 +6876,9 @@ spec:
                           description: 'Custom labels for the broker pods, example
                             use case: for Prometheus monitoring to capture the group
                             for each broker as a label, e.g.: kafka_broker_group:
-                            "default_group"'
+                            "default_group" these labels won''t override the reserved
```
Comment (amberimam) [inline charts/kafka-operator/templates/crds.yaml:6879]:
```suggestion
                            "default_group" these labels won't override the reserved
```

**Review by amberimam — CHANGES_REQUESTED** [review]
LGTM, just found a little typo.

File: charts/kafka-operator/templates/crds.yaml (hunk 1 above)
Comment (panyuenlau) [reply charts/kafka-operator/templates/crds.yaml:982]: @Kuvesz You will need to update this line: https://github.com/banzaicloud/koperator/blob/master/api/v1beta1/kafkacluster_types.go#L174, e.g. `won't` -> `will not`, and then run `make manifests`. 

It seems `controller-gen` does not like contracted forms and I forgot to run `make manifests` when I added this line

**Review by stoader — CHANGES_REQUESTED** [review]
Update the CRDs in https://github.com/banzaicloud/koperator/tree/master/config/test/crd/istio with CRDs picked from:
https://github.com/banzaicloud/istio-operator/blob/v2.13.0-rc.0/internal/assets/manifests/base/crds/crd-all.gen.yaml

https://github.com/banzaicloud/istio-operator/blob/v2.13.0-rc.0/deploy/charts/istio-operator/crds/istio-operator-crds.gen.yaml

File: config/samples/kafkacluster-with-istio.yaml [hunk 3]
```diff
@@ -8,7 +8,7 @@ spec:
   headlessServiceEnabled: false
   ingressController: "istioingress"
   istioControlPlane:
-    name: icp-v111x-sample
+    name: icp-v113x-sample
```
Comment (stoader) [inline config/samples/kafkacluster-with-istio.yaml:11]: I think we should add a comment here explaining that this should be the name of the `IstioControlPlane` custom resource and example of which can be found here https://github.com/banzaicloud/istio-operator/blob/v2.13.0-rc.0/config/samples/servicemesh_v1alpha1_istiocontrolplane.yaml

**Review by stoader — CHANGES_REQUESTED** [review]

**Review by stoader — APPROVED** [review]

**Review by bartam1 — COMMENTED** [review]
LGTM.
These can be updated as well to 113.
https://github.com/banzaicloud/koperator/blob/istio_113/controllers/tests/kafkacluster_controller_istioingress_test.go#L69
https://github.com/banzaicloud/koperator/blob/istio_113/controllers/tests/kafkacluster_controller_istioingress_test.go#L81
https://github.com/banzaicloud/koperator/blob/istio_113/controllers/tests/kafkacluster_controller_istioingress_test.go#L419

**Review by panyuenlau — APPROVED** [review]

**Review by bartam1 — APPROVED** [review]

**Pull request merged**